lib/omnibus/builder.rb

```diff
@@ -105,27 +108,101 @@ def command(command, options = {})
     #
     def make(*args)
       options = args.last.is_a?(Hash) ? args.pop : {}
-      command = args.join(' ')
 
-      make = if makebin = options.delete(:bin)
-               makebin
-             elsif Omnibus.which('gmake')
-               env = options.delete(:env) || {}
-               env = { 'MAKE' => 'gmake' }.merge(env)
-               options[:env] = env
-               'gmake'
-             else
-               'make'
-             end
+      make = options.delete(:bin) ||
+        # Prefer gmake on non-windows environments.
+        if !windows? && Omnibus.which('gmake')
+          env = options.delete(:env) || {}
+          env = { 'MAKE' => 'gmake' }.merge(env)
+          options[:env] = env
+          'gmake'
+        else
+          'make'
+        end
+
+      make_cmd = ([make] + args).join(' ').strip
+      if windows?
+        # The msys make that we deploy requires that we first invoke a shell.
+        # Fire up bash.
+        # TODO: Eventually, have make search through "build time dependencies"
+        # for bash/make instead.
+        bash_bin = embedded_msys_bin('bash.exe')
+        # TODO: I need to powershell/cmd escape command.strip.... :( No.
+        # We're going to use ' and hope for the best.
```

Comment:
You might be able to take advantage of the Ruby Standard Library `Shellwords.escape`:
http://ruby-doc.org/stdlib-2.0.0/libdoc/shellwords/rdoc/Shellwords.html#method-c-escape